providers/util.go

```diff
@@ -34,10 +34,16 @@ func makeOIDCHeader(accessToken string) http.Header {
 	return makeAuthorizationHeader(tokenTypeBearer, accessToken, extraHeaders)
 }
 
-func makeLoginURL(p *ProviderData, redirectURI, state string, extraParams url.Values) url.URL {
+func makeLoginURL(p *ProviderData, redirectURI, state, codeChallenge, codeChallengeMethod string, extraParams url.Values) url.URL {
```

Comment:
Adding this to extra params in the code code might make sense, that or we move everything into the providers package. At the moment I find it hard to follow when it's split across the main and providers package, though maybe that's unavoidable

Reply:
See https://github.com/oauth2-proxy/oauth2-proxy/pull/1541#discussion_r810565399

Would you like it in extra params then?